gen/generator.go

```diff
@@ -147,6 +147,16 @@ var templateFuncs = template.FuncMap{
 	"isReference":                           IsReference,
 	"getLegacyPropertyTestValue":            GetTestValue,
 	"getLegacyBlockTestValue":               GetBlockTestValue,
+	"getDeprecatedExplanation":              GetDeprecatedExplanation,
+	"getIgnoredExplanation":                 GetIgnoredExplanation,
+}
+
+func GetDeprecatedExplanation(attributeName, replacedByAttributeName string) string {
+	return fmt.Sprintf("Attribute '%s' is deprecated, please refer to '%s' instead. The attribute will be removed in the next major version of the provider.", attributeName, replacedByAttributeName)
+}
+
+func GetIgnoredExplanation(attributeName string) string {
+	return fmt.Sprintf("Attribute `%s` is deprecated. The configuration was not functioning as intended because the Managed Object (MO) created by the pre-migrated resource was either configured incorrectly or exposed without any implemented functionality on the APIC. The MO for this attribute is no longer created on the APIC, but the existing MO will remain present until the resource is destroyed. This attribute will be removed in the next major version of the provider.", attributeName)
```

Comment:
If the configuration was not functioning as intended then the below statement would never be a scenario, correct?
"but the existing MO will remain present until the resource is destroyed."